src/AWS.Deploy.CLI/Commands/CommandHandlerInput/CommandInputService.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+namespace AWS.Deploy.CLI.Commands.CommandHandlerInput
+{
+    public interface ICommandInputService
+    {
+        bool Diagnostics { get; }
+        DeleteCommandHandlerInput? DeleteInput { get; set; }
+        DeployCommandHandlerInput? DeployInput { get; set; }
+        GenerateDeploymentProjectCommandHandlerInput? GenerateDeploymentProjectInput { get; set; }
+        ListCommandHandlerInput? List { get; set; }
+        ServerModeCommandHandlerInput? ServerModeInput { get; set; }
+    }
+
+    public class CommandInputService : ICommandInputService
```

Comment:
This is weird. A base class that has a property to each of the sub classes. Is the goal to have input that has some global settings like `Diagnostics` and the command specific settings?

If so I would recommend using generics instead of a union of all of the children. Something like

```
    public interface ICommandInputService
    {
        bool Diagnostics { get; }
    }

    public interface ICommandInputService<T> : ICommandInputService
    {
        T Input { get; }
    }

    public CommandInputService<T> : ICommandInputService<T>
    {
          public bool Diagnostics => ...
          T Input {get;set;}
    }
```

Reply:
I'm not convinced we need `ICommandInputService` especially if we the Diagnostics flag into the interactive services. All I see it doing is passing around Diagnostics and the input type. But the input shape isn't even used once the command is executed. The input types are currently just used to translate the CLI switches into the arguments of the Execute commands then throw away. This change keeps them around for no new reason.

Reply:
Removed `ICommandInputService` for now, at this point not needed.